packages/main/src/themes/TextArea.css

```diff
@@ -166,158 +259,155 @@
+:host([value-state="Error"]:not([readonly]):not([disabled])) .ui5-textarea-inner,
+:host([value-state="Warning"]:not([readonly]):not([disabled])) .ui5-textarea-inner,
+:host([value-state="Information"]:not([readonly]):not([disabled])) .ui5-textarea-inner {
+	border-style: none;
 }
 
-:host([value-state="Error"]:not([readonly])) .ui5-textarea-inner {
-	font-weight: var(--_ui5_input_error_font_weight);
-	background: var(--sapField_InvalidBackgroundStyle);
-	background-color: var(--sapField_InvalidBackground);
+:host([value-state="Warning"]:not([readonly]):not([disabled])) .ui5-textarea-wrapper {
+	background: var(--sapField_WarningBackgroundStyle);
+	background-color: var(--sapField_WarningBackground);
+	border-color: var(--_ui5_textarea_value_state_warning_border_color);
 }
 
-:host([value-state="Error"]:not([readonly]):not([focused]):hover) .ui5-textarea-inner {
-	box-shadow: var(--sapContent_Negative_Shadow);
-	background-color: var(--_ui5_textarea_error_hover_background_color);
+:host([value-state="Warning"][focused]:not([readonly])) .ui5-textarea-wrapper {
+	background-color: var(--_ui5_textarea_focused_value_state_warning_background);
+	border-color: var(--_ui5_textarea_focused_value_state_warning_border_color);
 }
 
-:host([value-state="Error"][focused]:not([readonly])) .ui5-textarea-inner {
-	background-image: none;
-	box-shadow: var(--_ui5_textarea_value_state_error_focus_box_shadow);
-	background-color: var(--_ui5_textarea_error_focused_background_color);
+:host([value-state="Warning"][focused]:not([readonly])) .ui5-textarea-wrapper:after {
+	border-color: var(--_ui5_textarea_focused_value_state_warning_focus_outline_color);
 }
 
-:host([value-state="Information"]:not([readonly])) .ui5-textarea-inner {
-	background: var(--sapField_InformationBackgroundStyle);
-	background-color: var(--sapField_InformationBackground);
+:host([value-state="Warning"]:not([readonly])) .ui5-textarea-wrapper::before{
+	background-color: var(--_ui5_textarea_value_state_warning_border_botom_color);
 }
 
-:host([value-state="Information"]:not([readonly]):not([focused]):hover) .ui5-textarea-inner {
-	box-shadow: var(--sapContent_Informative_Shadow);
+:host([value-state="Warning"]:not([readonly]):not([focused]):hover) .ui5-textarea-wrapper {
 	background-color: var(--sapField_Hover_Background);
+	box-shadow: var(--_ui5_textarea_value_state_warning_hover_box_shadow);
 }
 
-:host([value-state="Information"][focused]:not([readonly])) .ui5-textarea-inner {
-	background-image: none;
-	box-shadow: var(--_ui5_textarea_focus_box_shadow);
-	background-color: var(--sapField_Focus_Background);
-}
-
-:host([value-state="Success"]:not([readonly])) .ui5-textarea-inner {
+:host([value-state="Success"]:not([readonly]):not([disabled])) .ui5-textarea-wrapper {
 	background: var(--sapField_SuccessBackgroundStyle);
 	background-color: var(--sapField_SuccessBackground);
+	border-color: var(--_ui5_textarea_value_state_success_border_color);
```

Comment:
border-color: var(--_ui5_textarea_value_state_success_border_color); do we need a parameter, it is used only here with value var(--sapField_SuccessColor); for all supported themes